catalyst-gateway/bin/src/service/api/documents/mod.rs

```diff
@@ -73,15 +78,15 @@ impl DocumentApi {
         &self, /// The document to PUT
         document: Cbor<Body>,
         /// Authorization required.
-        _auth: CatalystRBACSecurityScheme,
+        _auth: NoneOrRBAC,
     ) -> put_document::AllResponses {
         match document.0.into_bytes_limit(MAXIMUM_DOCUMENT_SIZE).await {
-            Ok(document) => put_document::endpoint(document).await,
+            Ok(doc_bytes) => put_document::endpoint(doc_bytes.to_vec()).await,
             Err(ReadBodyError::PayloadTooLarge) => put_document::Responses::PayloadTooLarge.into(),
-            Err(_err) => {
-                put_document::Responses::BadRequest(Json(PutDocumentBadRequest::new(
-                    "Failed to read document from the request",
-                )))
+            Err(e) => {
+                put_document::Responses::BadRequest(Json(PutDocumentBadRequest::new(&format!(
+                    "Failed to read document from the request, err: {e}"
```

Comment:
Never serialize Err into a 400 response.  Produce a properly formatted response (which can not have the `Err` in it anywhere).